pkg/client/automation/client.go

```diff
@@ -116,7 +116,7 @@ func WithClientRequestLimiter(limiter *concurrency.Limiter) func(client *Client)
 // If none is set, the default Monaco CLI specific user-agent is sent.
 func WithCustomUserAgentString(userAgent string) func(client *Client) {
 	return func(d *Client) {
-		d.client = &http.Client{Transport: useragent.NewCustomUserAgentTransport(d.client.Transport, userAgent)}
+		d.client = rest.NewRestClient(&http.Client{Transport: useragent.NewCustomUserAgentTransport(d.client.Client().Transport, userAgent)}, rest.CreateRateLimitStrategy())
```

Comment:
this will overwrite the rate-limiting strategy as well as the traffic logger that were previously set for the `d.client`

you currently don't export any of the fields, but I think for options like this one, we still only want to replace the http.Client inside the rest.Client (or create a new one that re-uses the ratelimit and logger that was in the client before the option applied)